tests/lib/rules/function-component-definition.js

```diff
@@ -166,6 +166,16 @@ ruleTester.run('function-component-definition', rule, {
     code: 'function Hello(props): ReactNode { return <p/> }',
     options: [{namedComponents: 'function-declaration'}],
     parser: parsers.TYPESCRIPT_ESLINT
+  }, {
+    // https://github.com/yannickcr/eslint-plugin-react/issues/2765
+    code: [
+      'const obj = {',
+      '  serialize(el) {',
+      '    return <p/>',
+      '  }',
+      '}'
+    ].join('\n'),
+    options: [{unnamedComponents: 'arrow-function'}]
```

Comment:
could we also add test cases with a bunch of other options, to be clear that no concise object method will ever be detected as a component?